### PR TITLE
Add enhancements template and script for table of contents

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Add an enhancements template, a trimmed version from k/enhancements.

Also copy the scripts to update and verify the TOC

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: hack/update-toc.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+export PATH="${TMP_DIR}:${PATH}"
+cd "${ROOT}"
+
+# Update tables of contents if necessary.
+find keps -name '*.md' \
+    | grep -Fxvf hack/.notableofcontents \
+    | xargs mdtoc --inplace --max-depth=5  || (
+      echo "Failed generating TOC. If this failed silently and you are on mac, try 'brew install grep'"
+      exit 1
+    )
```
Comment (kerthcet) [inline hack/update-toc.sh:52]: Add a new line at the end? Similar to other files.

Comment (alculquicondor) [reply hack/update-toc.sh:52]: Done

File: Makefile [hunk 2]
```diff
@@ -113,6 +113,10 @@ gomod-verify:
 	$(GO_CMD) mod tidy
 	git --no-pager diff --exit-code go.mod go.sum
 
+.PHONY: toc-verify
+toc-verify:
+	./hack/verify-toc.sh
+
 .PHONY: vet
```
Comment (kerthcet) [inline Makefile:120]: Should we also add `update-toc` to run the update-toc.sh.

Comment (alculquicondor) [reply Makefile:124]: Done

**Comment by kerthcet** [comment]
Inspired by k/enhancements

/lgtm

**Pull request merged**